### PR TITLE
fix a bug that causes solr updates to fail

### PR DESCRIPTION
**Pull request by peetucket**
@lmcglohon

### CODE REVIEW

**Comment by lmcglohon** [comment]
@peetucket Looks good to me!

File: lib/writer/solr_client.rb [hunk 1]
```diff
@@ -83,7 +83,7 @@ def self.update_solr_doc(id, solr_doc, solr_connector)
 
         params = "[{\"id\":\"#{id}\","
         solr_doc.each do |field_name, new_values|
-          next unless field_name == :id
+          next unless field_name != :id
```
Comment (cbeer) [inline lib/writer/solr_client.rb]: Can we rewrite this: `next if field_name == :id`?

Comment (jkeck) [inline lib/writer/solr_client.rb]: ~~Would this be more clear if written as `next if field_name == :id`?~~

me right now

Comment (peetucket) [inline lib/writer/solr_client.rb]: :smile:

**Comment by peetucket** [comment]
added a real test for updates (previous one was useless)
improved the other tests for add and deleting
some refactoring

@lmcglohon

**Comment by lmcglohon** [comment]
@peetucket released new gem.

**Pull request merged**